### PR TITLE
expose http and tcp log format

### PR DESCRIPTION
**Pull request by from-nibly**
expose `http-log-format` and `tcp-log-format` to set the `log-format` on the respectively.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,6 +160,18 @@ Configure HSTS - HTTP Strict Transport Security.
 
 https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
 
+### http-log-format
+
+Customize http the log format using log format varialbes. Default to the HAProxy default log format
```
Comment (coryfklein) [inline README.md]: "the http log format" and "variables"

File: README.md [hunk 2]
```diff
@@ -158,6 +160,18 @@ Configure HSTS - HTTP Strict Transport Security.
...
+
+https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#8.2.4
+
+### tcp-log-format
+
+Customize the tcp log format using log format varialbes. Default to the HAProxy default log format
```
Comment (coryfklein) [inline README.md]: "variables"

File: rootfs/etc/haproxy/template/haproxy.tmpl [hunk 3]
```diff
@@ -127,6 +127,9 @@ frontend httpfront
 {{ end }}
 {{ end }}
     default_backend {{ $ing.DefaultServer.RootLocation.Backend }}
+{{ if ne $cfg.HTTPLogFormat "" }}
+    log-format {{ $cfg.HTTPLogFormat }}
+{{ end }}
```
Comment (jcmoraisjr) [inline rootfs/etc/haproxy/template/haproxy.tmpl]: Move all the `log-format` to within the `if syslog` condition in the same group. If there are no syslog endpoint, `log-format` has no effect.

File: rootfs/etc/haproxy/template/haproxy.tmpl [hunk 4]
```diff
@@ -208,6 +217,9 @@ frontend httpsfront-{{ $host }}
 backend httpsback-default-backend
     mode tcp
     server {{ $host }} unix@/var/run/haproxy-{{ $host }}.sock send-proxy-v2
+{{ if ne $cfg.TCPLogFormat "" }}
+    log-format {{ $cfg.TCPLogFormat }}
+{{ end }}
```
Comment (jcmoraisjr) [inline rootfs/etc/haproxy/template/haproxy.tmpl]: Backends don't use log-format.

File: rootfs/etc/haproxy/template/haproxy.tmpl [hunk 5]
```diff
@@ -256,3 +274,6 @@ frontend healthz
     bind *:{{ $cfg.HealthzPort }}
     mode http
     monitor-uri /healthz
+{{ if ne $cfg.HTTPLogFormat "" }}
+    log-format {{ $cfg.HTTPLogFormat }}
+{{ end }}
```
Comment (jcmoraisjr) [inline rootfs/etc/haproxy/template/haproxy.tmpl]: Health-check url doesn't log so this can also be removed.

**Comment by from-nibly** [comment]
@jcmoraisjr Thanks for the suggestions! I have changed my pr.

**Comment by jcmoraisjr** [comment]
Thanks!

**Pull request merged**